mantidproject/mantid
# Refactor removing peaks from edge of detector

Removed edge peaks from FindPeaksMD.  Also refactored edgePixels function to not have 5 copies.

**To test:**
Data is in SystemTestData.  Show Instrument for TOPAZ_3132_event.  Then drag event_peaks0 and event_peaks30 into plot and check that event_peaks30 does not have peaks at edge.  Repeat with histo_peaks.
```python
Load(Filename='TOPAZ_3132_event.nxs', OutputWorkspace='TOPAZ_3132_event')
FilterBadPulses(InputWorkspace='TOPAZ_3132_event', OutputWorkspace='TOPAZ_3132_event', LowerCutoff=25)
ConvertToMD(InputWorkspace='TOPAZ_3132_event', QDimensions='Q3D', dEAnalysisMode='Elastic', Q3DFrames='Q_sample', LorentzCorrection=True, OutputWorkspace='TOPAZ_3132_md', MinValues='-25,-25,-25', MaxValues='25,25,25', SplitInto='2', SplitThreshold=50, MaxRecursionDepth=13, MinRecursionDepth=7)
FindPeaksMD(InputWorkspace='TOPAZ_3132_md', PeakDistanceThreshold=0.40000000000000002, MaxPeaks=50, OutputWorkspace='event_peaks30', EdgePixels=30)
FindPeaksMD(InputWorkspace='TOPAZ_3132_md', PeakDistanceThreshold=0.40000000000000002, MaxPeaks=50, OutputWorkspace='event_peaks0')

BinMD(InputWorkspace='TOPAZ_3132_md', AlignedDim0='Q_sample_x,-15,15,400', AlignedDim1='Q_sample_y,-15,15,400', AlignedDim2='Q_sample_z,-15,15,400', OutputWorkspace='TOPAZ_3132_histo')
FindPeaksMD(InputWorkspace='TOPAZ_3132_histo', PeakDistanceThreshold=0.40000000000000002, MaxPeaks=50, OutputWorkspace='histo_peaks30', EdgePixels=30)
FindPeaksMD(InputWorkspace='TOPAZ_3132_histo', PeakDistanceThreshold=0.40000000000000002, MaxPeaks=50, OutputWorkspace='histo_peaks0')
```

Fixes #19461 



---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
